hypervisor/base/types.go

```diff
@@ -22,6 +22,8 @@ type Hypervisor interface {
 
 	Modify(vmName string, cpus int, mem int) error
 
+	HealthCheck(vmName, property string) (*string, error)
```

Comment:
I think healthcheck here is an implementation detail.

I cannot see it as a part of the hypervisor interface.

Reply:
I have implemented this as a hypervisor service since i feel we would require checking health of the vm at multiple stages, like if we were to simulate a Disaster recovery drill on a vm cluster then the vm would go down many times and extending health check as a vermin command would be useful

Reply:
In such case I would go for the signature: 
`HealthCheck(vmName) error`